### PR TITLE
feat(config): new user configuration api for settings center

### PR DESCRIPTION
**Pull request by yizhouxw**
#### What type of PR is this?

type-feature

#### What this PR does / why we need it:
refactor user configuration API for
- support new features in ODC 4.2.4
- make this module easy to maintian

the ogranzation configuration related code was removed due not used after ODC 4.2.0.
for user configuration, previous version implementation was totally removed also, for simplify code.

the matadb table odc_user_configuration is repalced with new table config_user_configuration,
however the history user configuraiton migrated in via sql script.

the new version API design involves ConfigurationMeta, which is maintained in yml file.

#### Which issue(s) this PR fixes:
#1616 

#### Special notes for your reviewer:
add a basic authentication mechanism, off cause disabled by default.

developer can enable basic authentication by update metadb
update config_system_configuration set `value`='true' where `key`='odc.web.security.basic-authentication.enabled';

it will be useful for use curl to test API.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-migrate/src/main/resources/migrate/common/V_4_2_4_11__config_user_configuration.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  UNIQUE KEY `uk_user_configuration_user_id_key` (`user_id`, `key`)
+);
+
+-- migrate data from config_user to config_user_configuration
+insert into config_user_configuration(user_id, `key`, `value`, create_time, update_time, description)
+ select user_id, `key`,`value`, create_time, update_time, description from odc_user_configuration
+ where `key` in ('sqlexecute.defaultDelimiter', 'sqlexecute.defaultObjectDraggingOption',
+  'sqlexecute.defaultQueryLimit', 'sqlexecute.mysqlAutoCommitMode', 'sqlexecute.oracleAutoCommitMode');
+
+update config_user_configuration set `key` = 'odc.sqlexecute.default.delimiter' where `key` = 'sqlexecute.defaultDelimiter';
```
Comment (yhilmare) [inline server/odc-migrate/src/main/resources/migrate/common/V_4_2_4_11__config_user_configuration.sql:20]: sqlexecute in `config_system_configuration` is written as 'sq;-execute', should we follow this in `config_user_configuration`?

Comment (yizhouxw) [reply server/odc-migrate/src/main/resources/migrate/common/V_4_2_4_11__config_user_configuration.sql:20]: after discuss with @HSunboy , we will keep sqlexecute, as all last part use lowercaseCamel

File: server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigMetaService.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import com.oceanbase.odc.service.config.model.ConfigurationMeta;
+
+@Service
+public class UserConfigMetaService {
+    private Map<String, ConfigurationMeta> keyToConfigMetaMap = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        List<ConfigurationMeta> configurations =
+                YamlUtils.fromYamlList("user-config-meta.yml", ConfigurationMeta.class);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigMetaService.java:37]: add a sub dir? may be 'config-meta/user-config-meta.yml'

File: server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigService.java [hunk 3]
```diff
@@ -15,158 +15,108 @@
...
+@Service
 @Validated
-@SkipAuthorize("odc internal usage")
+@SkipAuthorize("isolated by user")
 public class UserConfigService {
-    /**
-     * data access object for user configuration
-     */
+    @Autowired
+    private SystemConfigService systemConfigService;
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigService.java:48]: useless code

File: server/odc-service/src/main/java/com/oceanbase/odc/service/session/SessionProperties.java [hunk 4]
```diff
@@ -126,6 +126,7 @@ public class SessionProperties {
      * 是否开启全链路诊断功能
      */
     @Value("${odc.session.full-link-trace.enabled:true}")
+    @Deprecated
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/session/SessionProperties.java:129]: prefer deleting

File: server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigMetaService.java (hunk 2 above)
Comment (yizhouxw) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/config/UserConfigMetaService.java]: move to config/meta dir now

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**